src/ir/operator.h

```diff
@@ -0,0 +1,52 @@
+//-----------------------------------------------------------------------------
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//----------------------------------------------------------------------------
+#ifndef SRC_IR_OPERATOR_H_
+#define SRC_IR_OPERATOR_H_
+
+#include <vector>
+
+#include "absl/container/flat_hash_map.h"
+#include "src/ir/types/type.h"
+#include "src/ir/value.h"
+
+namespace raksha::ir {
+
+class Block;
+
+// Signature of an operator. An operator could be simple operators (e.g., `+`,
+// `-`, `sql.select`, `sql.groupby`) or complex operators (e.g., particle,
```

Comment:
Probably should remove the things that we expect would have to be represented as `Block`s rather than `Operator`s from this list (eg, `sql.select` and `sql.groupby`).